app/services/ExactTargetService.scala

```diff
@@ -0,0 +1,126 @@
+package services
+
+import akka.agent.Agent
+import com.gu.membership.zuora.soap.Zuora.SubscribeResult
+import com.squareup.okhttp.Request.Builder
+import com.squareup.okhttp.{Response, MediaType, OkHttpClient, RequestBody}
+import com.typesafe.scalalogging.LazyLogging
+import configuration.Config
+import model.SubscriptionData
+import model.exactTarget.{ExactTargetException, SubscriptionDataExtensionRow}
+import play.api.libs.concurrent.Akka
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.libs.json._
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+trait ExactTargetService extends LazyLogging {
+  def etClient: ETClient
+
+  def sendETDataExtensionRow(subscribeResult: SubscribeResult, subscriptionData: SubscriptionData, zs: ZuoraService): Future[Unit] = {
+    val subscription = zs.subscriptionByName(subscribeResult.name)
+
+    val accAndPaymentMethod = for {
+      subs <- subscription
+      acc <- zs.account(subs)
+      pm <- zs.defaultPaymentMethod(acc)
+    } yield (acc, pm)
+
+    for {
+      subs <- subscription
+      rpc <- zs.normalRatePlanCharge(subs)
+      (acc, pm) <- accAndPaymentMethod
+      row = SubscriptionDataExtensionRow(
+        subscription = subs,
+        subscriptionData = subscriptionData,
+        ratePlanCharge = rpc,
+        paymentMethod = pm,
+        account = acc
+      )
+      response <- etClient.sendSubscriptionRow(row)
+    } yield {
+      response.code() match {
+        case 200 => {
+          logger.info(s"Successfully sent an email to confirm the subscription: $subscribeResult")
+        }
+        case _ => {
+          val errorMsg = s"Failed to send the subscription email $subscribeResult. Code: ${response.code()}, Message: ${response.body()}"
+          logger.error(errorMsg)
+          throw new ExactTargetException(errorMsg)
+        }
+      }
+    }
+  }
+}
+
+object ExactTargetService extends ExactTargetService {
+  lazy val etClient = ETClient
+}
+
+trait ETClient {
+  def sendSubscriptionRow(row: SubscriptionDataExtensionRow): Future[Response]
+}
+
+object ETClient extends ETClient with LazyLogging {
+  private val config = Config.ExactTarget
+  private val clientId = config.clientId
+  private val clientSecret = config.clientSecret
+  lazy val thankYouDataExtensionKey = Config.ExactTarget.thankYouDataExtensionKey
+  private val jsonMT = MediaType.parse("application/json; charset=utf-8")
+  private val httpClient = new OkHttpClient()
+  private val authEndpoint = "https://auth.exacttargetapis.com/v1/requestToken"
+  private val restEndpoint = "https://www.exacttargetapis.com/hub/v1"
+
+  private val accessToken = Agent(getAccessToken)
+  import play.api.Play.current
+
+  /**
+   * Get a token from Exact Target, that is expired every hour
+   * See https://code.exacttarget.com/apis-sdks/rest-api/using-the-api-key-to-authenticate-api-calls.html
+   * This call is blocking
+   */
+  private def getAccessToken: String = {
```

Comment:
This getAccessToken code has a fair bit in common with the Zuora (and also Membership) initialisation stuff, which @afiore has been refactoring with 'authSuppliers' in https://github.com/guardian/membership-frontend/commit/27ccc6b7958#diff-e10ac256e292cf1126561fcd015cb18c - that's only in Membership at the moment, but it's a common pattern we can probably apply to Subscription stuff as well soon. The `FutureSupplier` abstraction aims to cover dealing with the annoying start-up delays and auth-renewals code we've had to write several times.